config/initializers/remote_authenticatable.rb

```diff
@@ -0,0 +1,61 @@
+require 'devise/strategies/authenticatable'
+
+module Devise
+  module Strategies
+    class RemoteAuthenticatable < Authenticatable
+      def authenticate!
```

Comment:
This method, and a couple others, is quite long could we extract some of the behaviour into self-contained methods?

Reply:
I think this is done as part of the latest commit